operator/wgc.go

```diff
@@ -108,69 +113,108 @@ func (r *ClusterClientReconciler) Reconcile(ctx context.Context, c *v1beta.Wireg
 		return ctrl.Result{}, fmt.Errorf("error listing peers: %w", err)
 	}
 
-	for i, wg := range wgcs.Items {
+	// find out what node this client is on
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		return ctrl.Result{}, fmt.Errorf("NODE_NAME environment variable not set")
+	}
+
+	peers := []wgPeer{}
+	for _, wg := range wgcs.Items {
 		r.log.Info("WireguardClusterClient", "name", wg.Name)
 
 		updateWgc := false
-		skName := wg.Spec.PrivateKeySecretRef.Name
-		if skName == "" {
-			skName = "wgc-" + wg.Name
-			wg.Spec.PrivateKeySecretRef.Name = skName
-			updateWgc = true
+		node := v1beta.WireguardClusterClientNode{}
+		for _, n := range wg.Spec.Nodes {
+			if n.NodeName == nodeName {
+				node = n
+				break
+			}
 		}
-		skNamespace := GetK8sNamespace()
-		skKey := wg.Spec.PrivateKeySecretRef.Key
-		if skKey == "" {
-			skKey = "privateKey"
-			wg.Spec.PrivateKeySecretRef.Key = skKey
-			updateWgc = true
+		if node.NodeName == "" {
+			r.log.Warn("Node not found", "name", wg.Name, "node", nodeName)
```

Comment:
I think this should set a node label so that certain apps can use the existence of the wgc as a scheduling requirement. Also makes it possible to add a node and add the WGc item later.

Reply:
Or readyness thing even. I don't know what they're called. Taint? For example what cilium uses to mark nodes as cilium ready.

Reply:
A taint is roughly the inverse of a label; https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/.

The rough idea is that it can prevent pods from being scheduling when you can't tolerate said taint. eg

```
spec:
  tolerations:
    - key: "wga.kraudcloud.com/wgcNotReady"
      operator: "Exists"
      effect: "NoSchedule"
```

I think a label is more convenient, because the "default" state of a node isn't that it has a wgc pod running on it.

Rather than marking unready, marking ready would be better.

Reply:
That makes sense

Reply:
Uuuh but will removing a label actually deschedule the pod? 

Although I'm not sure if it should. A temporary crash in wgc shouldn't result in rescheduling I guess. 

Label probably good